Scripts/BuildPhases/LintAppLocalizedStringsUsage.swift

```diff
@@ -39,54 +39,61 @@ class Xcodeproj {
     }
-
-    /// The directory where the `.xcodeproj` resides. 
+    
+    /// The directory where the `.xcodeproj` resides.
     var projectDirectory: URL { projectURL.deletingLastPathComponent().deletingLastPathComponent() }
     /// The list of `PBXNativeTarget` targets in the project. Convenience getter for `PBXProjFile.nativeTargets`
     var nativeTargets: [PBXNativeTarget] { pbxproj.nativeTargets }
     /// The list of `PBXBuildFile` files a given `PBXNativeTarget` will build. Convenience getter for `PBXProjFile.buildFiles(for:)`
     func buildFiles(for target: PBXNativeTarget) -> [PBXBuildFile] { pbxproj.buildFiles(for: target) }
-
+    
     /// Finds the full path / URL of a `PBXBuildFile` based on the groups it belongs to and their `sourceTree` attribute
-    func resolveURL(to buildFile: PBXBuildFile) -> URL {
-        do {
-            let fileRef = try self.pbxproj.object(id: buildFile.fileRef) as PBXFileReference
-            return resolveURL(objectUUID: buildFile.fileRef, object: fileRef)
-        } catch {
-            // Cover `XCVersionGroup` (like `*.xcdatamodel`) and `PBXVariantGroup` (like `*.strings`)
-            let fileRef = try! self.pbxproj.object(id: buildFile.fileRef) as PBXGroup
-            return resolveURL(objectUUID: buildFile.fileRef, object: fileRef)
+    func resolveURL(to buildFile: PBXBuildFile) throws -> URL? {
+        if let fileRefID = buildFile.fileRef, let fileRefObject = try? self.pbxproj.object(id: fileRefID) as PBXFileReference {
+            return try resolveURL(objectUUID: fileRefID, object: fileRefObject)
+        } else {
+            // If the `PBXBuildFile` is pointing to `XCVersionGroup` (like `*.xcdatamodel`) and `PBXVariantGroup` (like `*.strings`)
+            // (instead of a `PBXFileReference`), then in practice each file in the group's `children` will be built by the Build Phase.
+            // In practice we can skip parsing those in our case and save some CPU, as we don't have a need to lint those non-source-code files.
+            return nil // just skip those (but don't throw — those are valid use cases in any pbxproj, just ones we don't care about)
         }
     }
-
+    
     /// Finds the full path / URL of a PBXReference (`PBXFileReference` of `PBXGroup`) based on the groups it belongs to and their `sourceTree` attribute
-    private func resolveURL<T: PBXReference>(objectUUID: ObjectUUID, object: T) -> URL {
+    private func resolveURL<T: PBXReference>(objectUUID: ObjectUUID, object: T) throws -> URL? {
         if objectUUID == self.pbxproj.rootProject.mainGroup { return URL(fileURLWithPath: ".", relativeTo: projectDirectory) }
-
+        
         switch object.sourceTree {
         case .absolute:
-            return URL(fileURLWithPath: object.path!)
+            guard let path = object.path else { throw ProjectInconsistencyError.incorrectAbsolutePath(id: objectUUID) }
+            return URL(fileURLWithPath: path)
```

Comment:
[Save those ponies](https://alisoftware.github.io/swift/2015/09/06/thinking-in-swift-1/)! 😄 🐴